src/test/scala/org/hammerlab/guacamole/LociSetSuite.scala

```diff
@@ -162,6 +160,13 @@ class LociSetSuite extends GuacFunSuite with Matchers {
     set1.union(set2).toString should equal("chr1:40-43")
   }
 
+  sparkTest("loci set parsing with contig lengths") {
+    LociSet.parse(
+      "chr1,chr2,17,chr2:3-5,chr20:10-20",
+      Some(Map("chr1" -> 10, "chr2" -> 20, "17" -> 12, "chr20" -> 5000))).toString should equal(
+        "17:0-12,chr1:0-10,chr2:0-20,chr20:10-20")
```

Comment:
The first `chr2` in the list is just a bare chromosome, which means "include the whole chromosome". The second `chr2` in the list is thereofore irrelevant -- we're taking the union of `chr2:0-20` and `chr2:3-5` and get back `chr2:0-20`. Does that make sense?